src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -314,13 +353,10 @@ protected virtual void OnDocumentClosing(DocumentId documentId)
         /// </summary>
         protected void ClearSolution()
         {
-            using (_serializationLock.DisposableWait())
-            {
-                var oldSolution = this.CurrentSolution;
-                this.ClearSolutionData();
-
-                this.RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind.SolutionCleared, oldSolution, this.CurrentSolution);
-            }
+            this.SetCurrentSolution(
+                oldSolution => this.CreateSolution(oldSolution.Id),
+                WorkspaceChangeKind.SolutionCleared,
+                onBeforeUpdate: (_, _) => this.ClearOpenDocuments());
```

Comment:
just inlined CLearSolutionData here.  since all it did was ClearOpenDocs, then create an empty solution, we only need the former here as the transformation already takes care of producing the latter.